packages/autocomplete-js/src/__tests__/autocomplete.test.ts

```diff
@@ -151,6 +155,90 @@ describe('autocomplete-js', () => {
     `);
   });
 
+  test("renders with an auto-incremented id if there's multiple instances", async () => {
+    const container = document.createElement('div');
+
+    document.body.appendChild(container);
+    autocomplete({
+      container,
+    });
+    autocomplete({
+      container,
+    });
+    autocomplete({
+      container,
+    });
+
+    await waitFor(() => {
+      expect(
+        document.querySelector('#autocomplete-0-label')
+      ).toBeInTheDocument();
+      expect(
+        document.querySelector('#autocomplete-1-label')
+      ).toBeInTheDocument();
+      expect(
+        document.querySelector('#autocomplete-2-label')
+      ).toBeInTheDocument();
+    });
+  });
+
+  test("doesn't increment the id when toggling detached mode", async () => {
+    const container = document.createElement('div');
+
+    document.body.appendChild(container);
+    const { update } = autocomplete<{ label: string }>({
+      container,
+    });
+
+    await waitFor(() => {
+      expect(
+        document.querySelector('.aa-DetachedSearchButton')
+      ).not.toBeInTheDocument();
+      expect(
+        document.querySelector('#autocomplete-3-label')
```

Comment:
Where does this `3` come from? Previous tests? If so, a change in another tests will make this one fail, which is too fragile.

Another solution would be to mock `generateAutocompleteId` and to make sure that it's not called when toggling Detached mode.

Reply:
You can use the same strategy as this: https://github.com/algolia/autocomplete/blob/b397149ee785b84f7bd070009bf1e13f8799c56a/packages/autocomplete-js/src/__tests__/getAlgoliaHits.test.ts#L7-L14

Reply:
Fixed here: https://github.com/algolia/autocomplete/pull/489/commits/38bc0a7942f72d977c30eca10c7760c8ebb3c217